### PR TITLE
doc: Remove unused section from release process

### PR DESCRIPTION
**Pull request by maflcko**
The section does not apply to us, because it is being done by the maintainers of the `bitcoin.org` website, see https://github.com/bitcoin-dot-org/Bitcoin.org/issues/3641#issuecomment-819586576 . Documentation about the process is in their repo: https://github.com/bitcoin-dot-org/bitcoin.org/blob/master/docs/adding-events-release-notes-and-alerts.md#release-notes

### CODE REVIEW

**Review by promag — COMMENTED** [review]
Concept ACK.

File: doc/release-process.md
Comment (jarolrod) [inline doc/release-process.md:299]: ack on changing the links to the `bitcoincore.org` site. 

The binaries still end up on the `bitcoin.org` site, can we have something that documents the how they end up there? (the policy with dealing with `bitcoin.org`)

**Review by jarolrod — COMMENTED** [review]
ACK fa051d3e5e50fb5726fc25464fef348255599e1e

But maybe mention the policy with dealing with `bitcoin.org`. For example, a small sentence that points to their release notes document

**Comment by MarcoFalke** [comment]
>  For example, a small sentence that points to their release notes document

It is really up to them to decide the process and whether to document it. Putting a link to https://github.com/bitcoin-dot-org/bitcoin.org/blob/master/docs/adding-events-release-notes-and-alerts.md#release-notes is possible, but will need maintenance if it is changed. Is there a specific need to link to the process of an external site?

**Comment by jarolrod** [comment]
If we don't actively take any steps to get the release on the bitcoin.org website, then this change is fine as is

**Review by Zero-1729 — APPROVED** [review]
ACK fa051d3e5e50fb5726fc25464fef348255599e1e

**Comment by Rspigler** [comment]
Concept ACK, this is out of date and no need to document policies of external websites.

**Comment by jarolrod** [comment]
> Concept ACK, this is out of date and no need to document policies of 3rd party websites.

Our binaries end up on the bitcoin.org site, this means it is technically a part of our release process, no? 

I would like there to be at least a note on this. At minimum this note can be left for a follow up as to not hold this change up.

**Comment by sipa** [comment]
They're placed on bitcoin.org by the maintainers of that website. It's not a step that Bitcoin Core's maintainers are involved in.

**Review by fanquake — APPROVED** [review]
ACK fa051d3e5e50fb5726fc25464fef348255599e1e

**Comment by Rspigler** [comment]
@jarolrod see here: https://github.com/bitcoin-dot-org/Bitcoin.org/issues/3641#issuecomment-819387245

**Pull request merged**
